examples/kubernetes/connectivity-check/pod-to-c-intra-node-proxy-ingress-reject.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pod-to-c-intra-node-proxy-ingress-reject
+spec:
+  selector:
+    matchLabels:
+      name: pod-to-c-intra-node-proxy-ingress-reject
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: pod-to-c-intra-node-proxy-ingress-reject
+    spec:
+      containers:
+      - name: pod-to-c-intra-node-proxy-ingress-reject-container
+        image: docker.io/byrnedo/alpine-curl:0.1.8
+        command: ["/bin/ash", "-c", "sleep 1000000000"]
+        imagePullPolicy: IfNotPresent
+        livenessProbe:
+          exec:
+            command: ["ash", "-c", "! curl -sS --fail --connect-timeout 5 -o /dev/null echo-c/invalid-path"]
```

Comment:
I wonder if it would be better to check both allow and reject in a single yaml file since it is the same policy. I.e., do something like: `curl ... echo-c/ && ! curl ... echo-c/invalid-path`

Nit: `invalid-path` was a bit confusing for me because it was not clear that it would be served by the server. Maybe `/forbidden-path` is better?

Reply:
> I wonder if it would be better to check both allow and reject in a single yaml file since it is the same policy. I.e., do something like: curl ... echo-c/ && ! curl ... echo-c/invalid-path

Single YAML file or single deployment? The idea behind having them report separately in different pods is that we can more easily pinpoint what type of issue it is, rather than "policy doesn't work between these peers" it's "policy *allow* doesn't work between these peers" or "policy *reject* doesn't work between these peers".  That said if we're concerned about deploying too many pods then this could be an easy way to halve the number of these policy-related pods. The trick would be in how we ensure that the `livenessProbe` could easily reflect which of the two errors have occurred.

> Nit: invalid-path was a bit confusing for me because it was not clear that it would be served by the server. Maybe /forbidden-path is better?

Sure, I'll fix this up.

Reply:
> The idea behind having them report separately in different pods is that we can more easily pinpoint what type of issue it is

Another possibility brought up by @tgraf on Slack is to have multiple containers in the same pod.

Reply:
I've reimplemented with 2 containers per pod to test allow/deny L7 paths.

Reply:
Regarding the paths, it's `/public` and `/private` as those are the paths that the underlying server responds to.

Reply:
Thanks! FWIW, the server seems to also respond with empty messages on other paths, but `/public` and `/private` makes it unambiguous.